iphone/Classes/TiUIListView.m

```diff
@@ -531,22 +531,20 @@ -(void)buildResultsForSearchText
                 }
             }
             if (thisSection != nil) {
-                if ([thisSection count] > 0) {
-                    [_searchResults addObject:thisSection];
-                    
-                    if (sectionTitles != nil && sectionIndices != nil) {
-                        NSNumber* theIndex = [NSNumber numberWithInt:i];
-                        if ([sectionIndices containsObject:theIndex]) {
-                            id theTitle = [sectionTitles objectAtIndex:[sectionIndices indexOfObject:theIndex]];
-                            if (filteredTitles == nil) {
-                                filteredTitles = [[NSMutableArray alloc] init];
-                            }
-                            if (filteredIndices == nil) {
-                                filteredIndices = [[NSMutableArray alloc] init];
-                            }
-                            [filteredTitles addObject:theTitle];
-                            [filteredIndices addObject:NUMUINTEGER([_searchResults count] -1)];
+                [_searchResults addObject:thisSection];
```

Comment:
In current implementation we are giving option of edit while searching. Suppose there are 2 sections  (section1 , section2) in list view and we started search. Suppose we searched for an item of section2 and want to move it in section1. Suppose the keywords which we searched is only available in section2 and not in section1. So it should show section1 with 0 rows and section2 with relevant search results, so that item from section2 to section1 can be moved. Mentioned condition get removed to add section1 with 0 rows, in _searchResults.

In previous implementation, if no result found for searched keyword in section1, section1 was deleted. As edit option was not supported for search result previously, so it was not affecting anything. 
 if ([thisSection count] > 0) { ...}
Above condition was doing the stuff for removing the section if search result was not found in particular section.

See the 3rd use case ( Search with search bar and listview), which is having move option.